packages/taco/src/conditions/schemas/address-allowlist.ts

```diff
@@ -0,0 +1,38 @@
+import { EthAddressSchema } from '@nucypher/shared';
+import { ethers } from 'ethers';
+import { z } from 'zod';
+
+import { baseConditionSchema } from './common';
+
+export const AddressAllowlistConditionType = 'address-allowlist';
+
+export const addressAllowlistConditionSchema = baseConditionSchema
+  .extend({
+    conditionType: z.literal(AddressAllowlistConditionType),
+    addresses: z
+      .array(EthAddressSchema)
+      .min(1, 'At least one address must be provided')
+      .max(25, 'Maximum of 25 addresses allowed')
+      // Ensure all addresses are valid Ethereum addresses with proper checksum
+      .refine((addrs) => {
+        try {
+          return addrs.every((addr) => {
+            // Then verify it has the correct checksum (comparing to canonical form)
+            return addr === ethers.utils.getAddress(addr);
+          });
+        } catch (e) {
+          // If any error occurs during validation, fail the validation
+          return false;
+        }
+      }, 'All addresses must be properly checksummed')
```

Comment:
I think `EthAddressSchema` should already take care of this. If it doesn't, then fix its `refine` function instead of using a new one here.

Reply:
Thanks @derekpierre, I did not want to mess with the current implementation of `EthAddressSchema` because it will introduce a breaking change (some codes, that was working before this merge, might stop working). So that will require to raise the major version number of our library. i.e. from 0.7 to 1. Are we planning to do so already? Or do I miss something here?